src/components/LoadingSpinner.tsx

```diff
@@ -59,25 +75,48 @@ const LoadSpinnerIconContainer = styled.div<{ $fontSize: number }>`
   }}
 `;
 
+// Disable completion checkmark by forcing value to always be shown as a number %
+const progressFormatter = (percent?: number): ReactNode => {
+  if (percent === undefined) {
+    return "";
+  }
+  return `${percent}%`;
+};
+
 /**
  * Applies a loading spinner overlay on the provided children, which can be toggled on and off via props.
  */
 export default function LoadingSpinner(inputProps: PropsWithChildren<LoadingSpinnerProps>): ReactElement {
   const props = { ...defaultProps, ...inputProps } as PropsWithChildren<Required<LoadingSpinnerProps>>;
+
+  // Delay showing progress bar slightly; this fixes a visual bug where the loading spinner
+  // would flash the progress bar at 100% right as it vanished.
+  const showProgressBar = useDebounce(props.progress !== undefined && props.progress !== null, VANISH_DURATION_MS);
+
   return (
     <LoadingSpinnerContainer style={props.style}>
       <LoadingSpinnerOverlay $loading={props.loading}>
-        <Spin
-          indicator={
-            <LoadSpinnerIconContainer $fontSize={props.iconSize}>
-              {/* Make larger loading icon by multiple of them together */}
-              {/* TODO: Make custom loading spinner SVG? */}
-              <LoadingOutlined />
-              <LoadingOutlined rotate={90} />
-              <LoadingOutlined rotate={135} />
-            </LoadSpinnerIconContainer>
-          }
-        ></Spin>
+        {showProgressBar && props.progress !== undefined && props.progress !== null ? (
```

Comment:
showProgressBar already has a check for props.progress being defined, is it an issue with the debounce that makes you have to recheck of that there? either way, I would move these conditionals into showProgressBar, 
```JavaScript
const showProgressBar = useDebounce(props.progress !== undefined && props.progress !== null, VANISH_DURATION_MS)  &&  props.progress !== undefined && props.progress !== null;
```

Reply:
Unfortunately for some reason TypeScript doesn't catch this... I ended up just adding check to cast `null` values to `undefined`, even though both values should never reach the offending line.